azure/scope/cluster.go

```diff
@@ -265,6 +265,35 @@ func (s *ClusterScope) LBSpecs() []azure.ResourceSpecGetter {
 		},
 	}
 
+	if s.APIServerLB().Type == infrav1.Public || s.APIServerLB().Type == "" {
```

Comment:
I think at this point the simplest, most correct condition for this is `if s.APIServerLB().Type != infrav1.Internal`. That's more consistent w/ the other changes, too.